packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -0,0 +1,205 @@
+        if (targets.length === 0 && fieldKind.multiplicity !== Multiplicity.Sequence) {
+            targets.push(this.createEmptyTarget());
+        }
+        return proxifyField(this, getFieldKind(rootSchema), targets);
+    }
+
+    public createEmptyTarget(): ProxyTarget {
+        if (!this.emptyNode) {
+            const emptyCursor = this.forest.allocateCursor();
+            this.emptyNode = new ProxyTarget(this, emptyCursor);
+            emptyCursor.free();
+        }
+        return this.emptyNode;
+    }
+
+    public createTarget(cursor: ITreeSubscriptionCursor): ProxyTarget {
+        const rootedCursor = cursor as unknown as RootedTextCursor;
+        let current = rootedCursor.getPath();
+        const upPath: string[] = [stringifyPath(current)];
+        while (current.parent !== undefined) {
+            current = current.parent;
+            upPath.unshift(stringifyPath(current));
+        }
+        const nodePath = upPath.join("/");
+        if (!this.nodes.has(nodePath)) {
+            this.nodes.set(nodePath, new ProxyTarget(this, cursor));
+        }
+        return this.nodes.get(nodePath)! as ProxyTarget;
+    }
+
+    private handleAfterChange(): void {
+        for (const [nodePath, target] of this.nodes) {
+            const path = this.tree?.locate(target.getAnchor())
+            if (path === undefined) {
+                target.free();
+                this.nodes.delete(nodePath);
+            }
+        }
+        for (const afterHandler of this.afterHandlers) {
+            afterHandler.call(this);
+        }
+    }
+
+    public registerAfterHandler(afterHandler: EditableTreeContextHandler): void {
+        this.afterHandlers.add(afterHandler);
+    }
+
+    public setNodeValue(path: NodePath, value: unknown): boolean {
+        return this.runTransaction((editor) => editor.setValue(path, value));
+    }
+
+    public insertNode(path: NodePath, nodeCursor: ITreeCursor): boolean {
+        return this.runTransaction((editor) => editor.insert(path, nodeCursor));
+    }
+
+    public deleteNode(path: NodePath, count: number): boolean {
+        return this.runTransaction((editor) => editor.delete(path, count));
+    }
+
+    private runTransaction(f: (editor: SequenceEditBuilder) => void): boolean {
+        assert(this.tree !== undefined, "Transaction-based editing requires `SharedTree` instance");
+        // this.prepareForEdit();
+        const result = this.tree?.runTransaction((forest: IForestSubscription, editor: SequenceEditBuilder) => {
+            f(editor);
+            return TransactionResult.Apply;
+        });
+        return result === TransactionResult.Apply;
+    }
+}
+
+/**
+ * A simple API for a Forest to interact with the tree.
+ *
+ * @returns {@link EditableTreeContext} which is used manage the cursors and anchors within the EditableTrees:
+ * This is necessary for supporting using this tree across edits to the forest, and not leaking memory.
```

Comment:
Missing documentation for the parameters (most importantly the contract around passing or not passing the `tree`).